crates/cheatnet/src/runtime_extensions/call_to_blockifier_runtime_extension/execution/entry_point.rs

```diff
@@ -129,41 +133,87 @@ pub fn execute_call_entry_point(
-                // right now,  each intermediate VM error is wrapped
-                // in a VirtualMachineExecutionErrorWithTrace  error
-                // with the stringified trace of all errors below
-                // it.
+                //   right now, each intermediate VM error is wrapped in a
+                //   VirtualMachineExecutionErrorWithTrace error with the stringified trace
+                //   of all errors below it.
+                //   When that's done, remove the 10000 character limitation.
+                let error_trace = context.error_trace();
                 EntryPointExecutionError::VirtualMachineExecutionErrorWithTrace {
-                    trace: context.error_trace(),
-                    source: error,
+                    trace: error_trace[..min(10000, error_trace.len())].to_string(),
+                    source: internal_error,
                 }
             }
-            other_error => other_error,
+            other_error => {
+                context
+                    .error_stack
+                    .push((storage_address, format!("{}\n", &other_error)));
+                other_error
+            }
         }
     });
 
-    let identifier = match entry_point.call_type {
-        CallType::Call => AddressOrClassHash::ContractAddress(entry_point.storage_address),
-        CallType::Delegate => AddressOrClassHash::ClassHash(entry_point.class_hash.unwrap()),
-    };
-    runtime_state.cheatnet_state.trace_data.exit_nested_call(
+    // region: Modified blockifier code
+    map_and_process_entry_point_call_result(
+        entry_point_call_result,
+        runtime_state,
         resources,
-        CallResult::from_execution_result(&result, &identifier),
-    );
+        entry_point,
+        context,
+    )
+    // endregion
+}
 
+fn map_and_process_entry_point_call_result(
+    result: EntryPointExecutionResult<(CallInfo, SyscallCounter)>,
+    runtime_state: &mut RuntimeState,
+    resources: &mut ExecutionResources,
+    entry_point: &mut CallEntryPoint,
+    context: &mut EntryPointExecutionContext,
+) -> EntryPointExecutionResult<CallInfo> {
     result
-    // region: Modified blockifier code
-    // We skip recursion depth decrease
-    // endregion
+        .map_err({
+            |error| {
+                runtime_state.cheatnet_state.trace_data.exit_nested_call(
+                    resources,
+                    &Default::default(),
+                    CallResult::from_err(
+                        &error,
+                        &AddressOrClassHash::get_by_entry_point(entry_point),
+                    ),
+                );
+                error
+            }
+        })
+        .map(|(call_info, this_call_syscall_counter)| {
+            let versioned_constants = context.tx_context.block_context.versioned_constants();
+            // We don't want the syscall resources to pollute the results
+            *resources -= &versioned_constants
```

Comment:
This is correct and I know I requested it, but I wonder if this won't get messy really quickly, as we are changing a global (in terms of execution) var that blockifier may rely on. Also the amount of business logic here is starting to get overwhelming.

Maybe we should store the vm resources that include the syscall costs in the tree and do the subtraction when converting to profiler structs - this way we can move some business logic to a separate place. Ofc extensive comments explaining what is happening would be appreciated if we do that. WDYT?

Reply:
Keep in mind that if we do it this way we shouldn't add the additional OS resources caused by syscalls in `calculate_used_gas` - we can then just remove the syscall counter from `UsedResources`

Reply:
Discussed offline, resolving since we want to keep it more clear for the user what the actual resource consumption is, and only include them in the gas calculation